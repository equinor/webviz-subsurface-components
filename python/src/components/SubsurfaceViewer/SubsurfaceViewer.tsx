import React from "react";
import { SubsurfaceViewerProps } from "@webviz/subsurface-viewer";

const SubsurfaceViewerComponent = React.lazy(
    () =>
        import(
            /* webpackChunkName: "webviz-subsurface-viewer" */ "@webviz/subsurface-viewer"
        )
);

const SubsurfaceViewer: React.FC<SubsurfaceViewerProps> = (props) => {
    return (
        <React.Suspense fallback={<div>Loading...</div>}>
            <SubsurfaceViewerComponent {...props} />
        </React.Suspense>
    );
};

SubsurfaceViewer.displayName = "SubsurfaceViewer";

export default SubsurfaceViewer;
