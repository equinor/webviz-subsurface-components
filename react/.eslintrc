{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends":  [
    "plugin:react/recommended",
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "plugins": ["@typescript-eslint"],
      "extends": [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended"
      ]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "amd": true,
    "browser": true,
    "es6": true,
    "jasmine": true,
    "node": true
  }
}
