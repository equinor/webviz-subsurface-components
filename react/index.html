<html>
    <head>
        <title>Demo application of webviz-subsurface-components</title>
    </head>
    <body>
        <div id='root'></div>
        <link rel="stylesheet" type="text/css" href="output.css">
        <script src="output.js"></script>
    </body>
</html>
