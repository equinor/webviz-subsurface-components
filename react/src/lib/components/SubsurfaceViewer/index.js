export { default, SubsurfaceViewerProps } from "./SubsurfaceViewer";
export {
    default as SubsurfaceViewerDashWrapper,
    SubsurfaceViewerDashWrapperProps,
} from "./SubsurfaceViewerDashWrapper";
