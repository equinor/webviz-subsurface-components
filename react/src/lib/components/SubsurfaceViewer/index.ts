export { default, SubsurfaceViewerProps } from "./SubsurfaceViewer";
export {
    default as DashSubsurfaceViewer,
    DashSubsurfaceViewerProps,
} from "./DashSubsurfaceViewer";
