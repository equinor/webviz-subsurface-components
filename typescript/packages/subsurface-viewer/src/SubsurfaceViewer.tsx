import React from "react";
import type { PickingInfo, Layer, LayersList } from "@deck.gl/core/typed";
import PropTypes from "prop-types";
import type { colorTablesArray } from "@emerson-eps/color-tables/";
import type { Unit } from "convert-units";
import convert from "convert-units";

import type { MjolnirGestureEvent } from "mjolnir.js";

import type {
    BoundsAccessor,
    MapMouseEvent,
    TooltipCallback,
    ViewStateType,
    ViewsType,
} from "./components/Map";

import { TGrid3DColoringMode } from "./layers/grid3d/grid3dLayer";

import Map, { createLayers } from "./components/Map";

export type {
    BoundsAccessor,
    MapMouseEvent,
    TooltipCallback,
    ViewStateType,
    ViewsType,
    colorTablesArray,
};

export { TGrid3DColoringMode };

export type LightsType = {
    headLight?: {
        intensity: number;
        color?: [number, number, number];
    };
    ambientLight?: {
        intensity: number;
        color?: [number, number, number];
    };
    pointLights?: [
        {
            intensity: number;
            position: [number, number, number];
            color?: [number, number, number];
        },
    ];

    directionalLights?: [
        {
            intensity: number;
            direction: [number, number, number];
            color?: [number, number, number];
        },
    ];
};

export type TLayerDefinition =
    | Record<string, unknown>
    | Layer
    | false
    | null
    | undefined;

/**
 * Properties of the SubsurfaceViewer component.
 */
export interface SubsurfaceViewerProps {
    id: string;
    resources?: Record<string, unknown>;
    /**
     * Array of externally created layers or layer definition records or JSON strings.
     * Add '@@typedArraySupport' : true in a layer definition in order to
     * use typed arrays as inputs.
     */
    layers?: TLayerDefinition[];

    bounds?: [number, number, number, number] | BoundsAccessor;
    cameraPosition?: ViewStateType | undefined;
    triggerHome?: number;

    views?: ViewsType;
    coords?: {
        visible?: boolean | null;
        multiPicking?: boolean | null;
        pickDepth?: number | null;
    };
    scale?: {
        visible?: boolean | null;
        incrementValue?: number | null;
        widthPerUnit?: number | null;
        cssStyle?: Record<string, unknown> | null;
    };
    coordinateUnit?: Unit;
    colorTables?: colorTablesArray;
    editedData?: Record<string, unknown>;
    setProps?: (data: Record<string, unknown>) => void;

    /**
     * Validate JSON datafile against schema
     */
    checkDatafileSchema?: boolean;

    /**
     * For get mouse events
     */
    onMouseEvent?: (event: MapMouseEvent) => void;

    getCameraPosition?: (input: ViewStateType) => void;

    /**
     * Will be called while layers are processed to rendered data.
     * @param progress vlaue between 0 and 100.
     */
    onRenderingProgress?: (progress: number) => void;

    onDragStart?: (info: PickingInfo, event: MjolnirGestureEvent) => void;
    onDragEnd?: (info: PickingInfo, event: MjolnirGestureEvent) => void;

    triggerResetMultipleWells?: number;
    /**
     * Range selection of the current well
     */
    selection?: {
        well: string | undefined;
        selection: [number | undefined, number | undefined] | undefined;
    };

    /**
     * Override default tooltip with a callback.
     */
    getTooltip?: TooltipCallback;

    lights?: LightsType;

    children?: React.ReactNode;

    /** A vertical scale factor, used to scale items in the view vertically */
    verticalScale?: number;
}

const SubsurfaceViewer: React.FC<SubsurfaceViewerProps> = ({
    id,
    resources,
    layers,
    bounds,
    cameraPosition,
    triggerHome,
    views,
    coords,
    scale,
    coordinateUnit,
    colorTables,
    editedData,
    setProps,
    checkDatafileSchema,
    onMouseEvent,
    selection,
    getTooltip,
    getCameraPosition,
    onRenderingProgress,
    onDragStart,
    onDragEnd,
    triggerResetMultipleWells,
    lights,
    children,
    verticalScale,
}: SubsurfaceViewerProps) => {
    // Contains layers data received from map layers by user interaction
    const [layerEditedData, setLayerEditedData] = React.useState(editedData);

    const [layerInstances, setLayerInstances] = React.useState<LayersList>([]);

    React.useEffect(() => {
        const enumerations: Record<string, unknown>[] = [];
        if (resources) enumerations.push({ resources: resources });
        if (editedData) enumerations.push({ editedData: editedData });
        else enumerations.push({ editedData: {} });

        if (!layers) {
            setLayerInstances([]);
            return;
        }
        const layersList = createLayers(layers, enumerations);
        setLayerInstances(layersList);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [layers]); // Note. Fixing this dependency list may cause infinite recursion.

    React.useEffect(() => {
        if (!editedData) return;

        setLayerEditedData({
            ...layerEditedData,
            ...editedData,
        });

        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [editedData]); // Note. Fixing this dependency list may cause infinite recursion.

    // This callback is used as a mechanism to update the component from the layers or toolbar.
    // The changes done in a layer, for example, are bundled into a patch
    // and sent to the parent component via setProps. (See layers/utils/layerTools.ts)
    const setEditedData = React.useCallback(
        (data: Record<string, unknown>) => {
            if (setProps == undefined) return;
            setProps({
                editedData: {
                    ...layerEditedData,
                    ...data,
                },
            });
        },
        [setProps, layerEditedData]
    );

    if (coordinateUnit && !convert().possibilities().includes(coordinateUnit)) {
        console.error(
            `Invalid coordinate unit: '${coordinateUnit}'. Valid units are: ${convert().possibilities()}`
        );
        coordinateUnit = undefined;
    }

    return (
        <Map
            id={id}
            layers={layerInstances}
            bounds={bounds}
            views={views}
            coords={coords}
            scale={scale}
            coordinateUnit={coordinateUnit as Unit}
            colorTables={colorTables}
            setEditedData={setEditedData}
            checkDatafileSchema={checkDatafileSchema}
            onMouseEvent={onMouseEvent}
            selection={selection}
            getTooltip={getTooltip}
            cameraPosition={cameraPosition}
            getCameraPosition={getCameraPosition}
            onRenderingProgress={onRenderingProgress}
            onDragStart={onDragStart}
            onDragEnd={onDragEnd}
            triggerHome={triggerHome}
            triggerResetMultipleWells={triggerResetMultipleWells}
            lights={lights}
            verticalScale={verticalScale}
        >
            {children}
        </Map>
    );
};

SubsurfaceViewer.defaultProps = {
    views: {
        layout: [1, 1],
        marginPixels: 0,
        showLabel: false,
        viewports: [{ id: "main-view", show3D: false, layerIds: [] }],
    },
    checkDatafileSchema: false,
};

SubsurfaceViewer.propTypes = {
    /**
     * The ID of this component, used to identify dash components
     * in callbacks. The ID needs to be unique across all of the
     * components in an app.
     */
    id: PropTypes.string.isRequired,

    /**
     * Resource dictionary made available in the DeckGL specification as an enum.
     * The values can be accessed like this: `"@@#resources.resourceId"`, where
     * `resourceId` is the key in the `resources` dict. For more information,
     * see the DeckGL documentation on enums in the json spec:
     * https://deck.gl/docs/api-reference/json/conversion-reference#enumerations-and-using-the--prefix
     */
    resources: PropTypes.objectOf(PropTypes.any),

    /* List of JSON object containing layer specific data.
     * Each JSON object will consist of layer type with key as "@@type" and
     * layer specific data, if any.
     */
    layers: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any).isRequired),

    /**
     * Coordinate boundary for the view defined as [left, bottom, right, top].
     * It can be either an array or a callback returning [number, number, number, number].
     */
    bounds: PropTypes.any,

    /**
     * Views configuration for map. If not specified, all the layers will be
     * displayed in a single 2D viewport.
     * Example:
     *      views = {
     *          "layout": [1, 1],
     *          "showLabel": false,
     *          "viewports": [
     *              {
     *                  "id": "view_1",
     *                  "name"?: "View 1"
     *                  "show3D"?: false,
     *                  "layerIds": ["layer-ids"],
     *                  "isSync?": true,
     *              }
     *          ]
     *      }
     */
    // TODO - define proper type for views prop
    views: PropTypes.any,

    /**
     * Parameters for the InfoCard component
     */
    coords: PropTypes.shape({
        /**
         * Toggle component visibility.
         */
        visible: PropTypes.bool,
        /**
         * Enable or disable multi picking. Might have a performance penalty.
         * See https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects
         */
        multiPicking: PropTypes.bool,
        /**
         * Number of objects to pick. The more objects picked, the more picking operations will be done.
         * See https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects
         */
        pickDepth: PropTypes.number,
    }),

    /**
     * Parameters for the Distance Scale component
     */
    scale: PropTypes.shape({
        /**
         * Toggle component visibility.
         */
        visible: PropTypes.bool,
        /**
         * Increment value for the scale.
         */
        incrementValue: PropTypes.number,
        /**
         * Scale bar width in pixels per unit value.
         */
        widthPerUnit: PropTypes.number,
        /**
         * Scale bar css style can be used for positioning.
         */
        cssStyle: PropTypes.objectOf(PropTypes.any),
    }),

    /**
     * Parameters for the Distance Scale component
     * Unit for the scale ruler
     */
    coordinateUnit: PropTypes.oneOf(convert().possibilities()),

    /**
     * Prop containing color table data
     */
    colorTables: PropTypes.array,

    /**
     * Prop containing edited data from layers
     */
    editedData: PropTypes.objectOf(PropTypes.any),

    /**
     * For reacting to prop changes
     */
    setProps: PropTypes.func,

    /**
     * Validate JSON datafile against schema
     */
    checkDatafileSchema: PropTypes.bool,
};

export default SubsurfaceViewer;
