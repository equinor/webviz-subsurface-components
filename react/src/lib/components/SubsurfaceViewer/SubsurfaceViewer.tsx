import { LayersList, Layer } from "@deck.gl/core/typed";
import Map, {
    ViewsType,
    TooltipCallback,
    ViewStateType,
    BoundsAccessor,
} from "./components/Map";
import { MapMouseEvent, jsonToObject } from "./components/Map";
import React from "react";
import PropTypes from "prop-types";
import { colorTablesArray } from "@emerson-eps/color-tables/";
import convert, { Unit } from "convert-units";

export interface SubsurfaceViewerProps {
    id: string;
    resources?: Record<string, unknown>;
    layers?: Record<string, unknown>[] | LayersList;
    bounds?: [number, number, number, number] | BoundsAccessor;
    views?: ViewsType;
    coords?: {
        visible?: boolean | null;
        multiPicking?: boolean | null;
        pickDepth?: number | null;
    };
    scale?: {
        visible?: boolean | null;
        incrementValue?: number | null;
        widthPerUnit?: number | null;
        cssStyle?: Record<string, unknown> | null;
    };
    coordinateUnit?: Unit;
    toolbar?: {
        visible?: boolean | null;
    };
    legend?: {
        visible?: boolean | null;
        cssStyle?: Record<string, unknown> | null;
        horizontal?: boolean | null;
    };
    colorTables?: colorTablesArray;
    editedData?: Record<string, unknown>;
    setProps?: (data: Record<string, unknown>) => void;

    /**
     * Validate JSON datafile against schema
     */
    checkDatafileSchema?: boolean;

    /**
     * For get mouse events
     */
    onMouseEvent?: (event: MapMouseEvent) => void;

    getCameraPosition?: (input: ViewStateType) => void;

    /**
     * If changed will reset camera to default position.
     */
    triggerHome?: number;
    triggerResetMultipleWells?: number;
    /**
     * Range selection of the current well
     */
    selection?: {
        well: string | undefined;
        selection: [number | undefined, number | undefined] | undefined;
    };

    /**
     * Override default tooltip with a callback.
     */
    getTooltip?: TooltipCallback;
    cameraPosition?: ViewStateType | undefined;

    children?: React.ReactNode;
}

const SubsurfaceViewer: React.FC<SubsurfaceViewerProps> = ({
    id,
    resources,
    layers,
    bounds,
    views,
    coords,
    scale,
    coordinateUnit,
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    legend,
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    toolbar,
    colorTables,
    editedData,
    setProps,
    checkDatafileSchema,
    onMouseEvent,
    selection,
    getTooltip,
    cameraPosition,
    getCameraPosition,
    triggerHome,
    triggerResetMultipleWells,
    children,
}: SubsurfaceViewerProps) => {
    // Contains layers data received from map layers by user interaction
    const [layerEditedData, setLayerEditedData] = React.useState(editedData);

    const [layerInstances, setLayerInstances] = React.useState<LayersList>([]);

    React.useEffect(() => {
        if (!layers) {
            setLayerInstances([]);
            return;
        }

        if (layers?.[0] instanceof Layer) {
            setLayerInstances(layers as LayersList);
            return;
        }

        const enumerations: Record<string, unknown>[] = [];
        if (resources) enumerations.push({ resources: resources });
        if (editedData) enumerations.push({ editedData: editedData });
        else enumerations.push({ editedData: {} });
        const layersList = jsonToObject(
            layers as Record<string, unknown>[],
            enumerations
        ) as LayersList;
        setLayerInstances(layersList);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [layers]); // Note. Fixing this dependency list may cause infinite recursion.
    React.useEffect(() => {
        if (!editedData) return;

        setLayerEditedData({
            ...layerEditedData,
            ...editedData,
        });

        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [editedData]); // Note. Fixing this dependency list may cause infinite recursion.

    // This callback is used as a mechanism to update the component from the layers or toolbar.
    // The changes done in a layer, for example, are bundled into a patch
    // and sent to the parent component via setProps. (See layers/utils/layerTools.ts)
    const setEditedData = React.useCallback(
        (data: Record<string, unknown>) => {
            if (setProps == undefined) return;
            setProps({
                editedData: {
                    ...layerEditedData,
                    ...data,
                },
            });
        },
        [setProps, layerEditedData]
    );

    if (coordinateUnit && !convert().possibilities().includes(coordinateUnit)) {
        console.error(
            `Invalid coordinate unit: '${coordinateUnit}'. Valid units are: ${convert().possibilities()}`
        );
        coordinateUnit = undefined;
    }

    return (
        <Map
            id={id}
            layers={layerInstances}
            bounds={bounds}
            views={views}
            coords={coords}
            scale={scale}
            coordinateUnit={coordinateUnit as Unit}
            colorTables={colorTables}
            setEditedData={setEditedData}
            checkDatafileSchema={checkDatafileSchema}
            onMouseEvent={onMouseEvent}
            selection={selection}
            getTooltip={getTooltip}
            cameraPosition={cameraPosition}
            getCameraPosition={getCameraPosition}
            triggerHome={triggerHome}
            triggerResetMultipleWells={triggerResetMultipleWells}
        >
            {children}
        </Map>
    );
};

SubsurfaceViewer.defaultProps = {
    views: {
        layout: [1, 1],
        showLabel: false,
        viewports: [{ id: "main-view", show3D: false, layerIds: [] }],
    },
    checkDatafileSchema: false,
};

SubsurfaceViewer.propTypes = {
    /**
     * The ID of this component, used to identify dash components
     * in callbacks. The ID needs to be unique across all of the
     * components in an app.
     */
    id: PropTypes.string.isRequired,

    /**
     * Resource dictionary made available in the DeckGL specification as an enum.
     * The values can be accessed like this: `"@@#resources.resourceId"`, where
     * `resourceId` is the key in the `resources` dict. For more information,
     * see the DeckGL documentation on enums in the json spec:
     * https://deck.gl/docs/api-reference/json/conversion-reference#enumerations-and-using-the--prefix
     */
    resources: PropTypes.objectOf(PropTypes.any),

    /* List of JSON object containing layer specific data.
     * Each JSON object will consist of layer type with key as "@@type" and
     * layer specific data, if any.
     */
    layers: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any).isRequired),

    /**
     * Coordinate boundary for the view defined as [left, bottom, right, top].
     * It can be either an array or a callback returning [number, number, number, number].
     */
    bounds: PropTypes.any,

    /**
     * Views configuration for map. If not specified, all the layers will be
     * displayed in a single 2D viewport.
     * Example:
     *      views = {
     *          "layout": [1, 1],
     *          "showLabel": false,
     *          "viewports": [
     *              {
     *                  "id": "view_1",
     *                  "name"?: "View 1"
     *                  "show3D"?: false,
     *                  "layerIds": ["layer-ids"],
     *                  "isSync?": true,
     *              }
     *          ]
     *      }
     */
    // TODO - define proper type for views prop
    views: PropTypes.any,

    /**
     * Parameters for the InfoCard component
     */
    coords: PropTypes.shape({
        /**
         * Toggle component visibility.
         */
        visible: PropTypes.bool,
        /**
         * Enable or disable multi picking. Might have a performance penalty.
         * See https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects
         */
        multiPicking: PropTypes.bool,
        /**
         * Number of objects to pick. The more objects picked, the more picking operations will be done.
         * See https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects
         */
        pickDepth: PropTypes.number,
    }),

    /**
     * Parameters for the Distance Scale component
     */
    scale: PropTypes.shape({
        /**
         * Toggle component visibility.
         */
        visible: PropTypes.bool,
        /**
         * Increment value for the scale.
         */
        incrementValue: PropTypes.number,
        /**
         * Scale bar width in pixels per unit value.
         */
        widthPerUnit: PropTypes.number,
        /**
         * Scale bar css style can be used for positioning.
         */
        cssStyle: PropTypes.objectOf(PropTypes.any),
    }),

    /**
     * Parameters for the Distance Scale component
     * Unit for the scale ruler
     */
    coordinateUnit: PropTypes.oneOf(convert().possibilities()),

    /**
     * @obsolete Toolbar should be added as annotation. This prop has no function.
     */
    toolbar: PropTypes.shape({
        /**
         * Toggle toolbar visibility
         */
        visible: PropTypes.bool,
    }),

    /**
     * @obsolete Legends should be added as annotations. This prop has no function.
     */
    legend: PropTypes.shape({
        /**
         * Toggle component visibility.
         */
        visible: PropTypes.bool,
        /**
         * Legend css style can be used for positioning.
         */
        cssStyle: PropTypes.objectOf(PropTypes.any),
        /**
         * Orientation of color legend
         */
        horizontal: PropTypes.bool,
    }),

    /**
     * Prop containing color table data
     */
    colorTables: PropTypes.array,

    /**
     * Prop containing edited data from layers
     */
    editedData: PropTypes.objectOf(PropTypes.any),

    /**
     * For reacting to prop changes
     */
    setProps: PropTypes.func,

    /**
     * Validate JSON datafile against schema
     */
    checkDatafileSchema: PropTypes.bool,

    /**
     * For get mouse events
     */
    onMouseEvent: PropTypes.func,
};

export default SubsurfaceViewer;
