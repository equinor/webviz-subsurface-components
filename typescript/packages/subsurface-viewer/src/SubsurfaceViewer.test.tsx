import { colorTables } from "@emerson-eps/color-tables";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import type { Unit } from "convert-units";
import "jest-styled-components";
import React from "react";
import mapData from "../../../../example-data/deckgl-map.json";
import SubsurfaceViewer from "./SubsurfaceViewer";

const colorTablesData = colorTables;

describe("Test Map component", () => {
    xit("snapshot test", () => {
        const { container } = render(
            <SubsurfaceViewer
                id={mapData[0].id}
                resources={mapData[0].resources}
                layers={mapData[0].layers}
                bounds={mapData[0].bounds as [number, number, number, number]}
                coords={mapData[0].coords}
                scale={mapData[0].scale}
                coordinateUnit={mapData[0].coordinateUnit as Unit}
                // @ts-expect-error TS2322
                legend={mapData[0].legend}
                editedData={mapData[0].editedData}
                views={{
                    layout: [1, 1],
                    viewports: [
                        {
                            id: "view_1",
                            show3D: false,
                            layerIds: [],
                        },
                    ],
                }}
                colorTables={[colorTablesData[0]]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    xit("snapshot test with edited data", () => {
        const { container } = render(
            <SubsurfaceViewer
                id={mapData[0].id}
                resources={mapData[0].resources}
                layers={mapData[0].layers}
                bounds={mapData[0].bounds as [number, number, number, number]}
                coords={mapData[0].coords}
                scale={mapData[0].scale}
                coordinateUnit={mapData[0].coordinateUnit as Unit}
                // @ts-expect-error TS2322
                legend={mapData[0].legend}
                editedData={{}}
                views={{
                    layout: [1, 1],
                    viewports: [
                        {
                            id: "view_1",
                            show3D: false,
                            layerIds: [],
                        },
                    ],
                }}
                colorTables={[colorTablesData[0]]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    xit("snapshot test with invalid array length", () => {
        const { container } = render(
            <SubsurfaceViewer
                id={mapData[0].id}
                resources={mapData[0].resources}
                layers={mapData[0].layers}
                // @ts-expect-error: Ignore a compile error for "bounds" prop for the sake of running a scenario
                bounds={[0, 0, 0]}
                coords={mapData[0].coords}
                scale={mapData[0].scale}
                coordinateUnit={mapData[0].coordinateUnit as Unit}
                // @ts-expect-error TS2339
                legend={mapData[0].legend}
                editedData={mapData[0].editedData}
                views={{
                    layout: [1, 1],
                    viewports: [
                        {
                            id: "view_1",
                            show3D: false,
                            layerIds: [],
                        },
                    ],
                }}
                colorTables={[colorTablesData[0]]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
