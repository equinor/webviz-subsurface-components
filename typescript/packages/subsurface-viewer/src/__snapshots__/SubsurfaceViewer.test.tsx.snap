// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Map component snapshot test 1`] = `
.c0 {
  display: inline-block;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

<div>
  <div
    id="DeckGL-Map-wrapper"
    style="position: absolute; z-index: 0; left: 0px; top: 0px; width: 100%; height: 100%;"
  >
    <canvas
      id="DeckGL-Map"
      style="left: 0px; top: 0px; width: 100%; position: absolute; height: 100%;"
    />
  </div>
  <div
    style="position: absolute; left: 10px; top: 10px;"
  >
    <label
      style="left: 10px; top: 10px;"
    >
      3
      km
    </label>
    <div
      style="width: 100px; height: 4px; border: 2px solid; display: inline-block; margin-left: 3px;"
    />
  </div>
  <div
    style="display: flex; align-items: flex-end; justify-content: right; position: absolute; height: 10%; width: 10%; bottom: 10px; right: 10px; z-index: 200;"
  >
    <div>
      <svg
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="c0"
        height="48"
        preserveAspectRatio="xMidYMid meet"
        role="progressbar"
        viewBox="24 24 48 48"
        width="48"
      >
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_13, rgba(222, 237, 238, 1))"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_100, rgba(0, 112, 121, 1))"
          stroke-dasharray="138.23007675795088"
          stroke-linecap="round"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
      </svg>
      <output
        class="c1"
      >
        Loading 0%
      </output>
      <br />
      Loading assets...
    </div>
  </div>
</div>
`;

exports[`Test Map component snapshot test with edited data 1`] = `
.c0 {
  display: inline-block;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

<div>
  <div
    id="DeckGL-Map-wrapper"
    style="position: absolute; z-index: 0; left: 0px; top: 0px; width: 100%; height: 100%;"
  >
    <canvas
      id="DeckGL-Map"
      style="left: 0px; top: 0px; width: 100%; position: absolute; height: 100%;"
    />
  </div>
  <div
    style="position: absolute; left: 10px; top: 10px;"
  >
    <label
      style="left: 10px; top: 10px;"
    >
      3
      km
    </label>
    <div
      style="width: 100px; height: 4px; border: 2px solid; display: inline-block; margin-left: 3px;"
    />
  </div>
  <div
    style="display: flex; align-items: flex-end; justify-content: right; position: absolute; height: 10%; width: 10%; bottom: 10px; right: 10px; z-index: 200;"
  >
    <div>
      <svg
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="c0"
        height="48"
        preserveAspectRatio="xMidYMid meet"
        role="progressbar"
        viewBox="24 24 48 48"
        width="48"
      >
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_13, rgba(222, 237, 238, 1))"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_100, rgba(0, 112, 121, 1))"
          stroke-dasharray="138.23007675795088"
          stroke-linecap="round"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
      </svg>
      <output
        class="c1"
      >
        Loading 0%
      </output>
      <br />
      Loading assets...
    </div>
  </div>
</div>
`;

exports[`Test Map component snapshot test with invalid array length 1`] = `
.c0 {
  display: inline-block;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

<div>
  <div
    id="DeckGL-Map-wrapper"
    style="position: absolute; z-index: 0; left: 0px; top: 0px; width: 100%; height: 100%;"
  >
    <canvas
      id="DeckGL-Map"
      style="left: 0px; top: 0px; width: 100%; position: absolute; height: 100%;"
    />
  </div>
  <div
    style="position: absolute; left: 10px; top: 10px;"
  >
    <label
      style="left: 10px; top: 10px;"
    >
      3
      km
    </label>
    <div
      style="width: 100px; height: 4px; border: 2px solid; display: inline-block; margin-left: 3px;"
    />
  </div>
  <div
    style="display: flex; align-items: flex-end; justify-content: right; position: absolute; height: 10%; width: 10%; bottom: 10px; right: 10px; z-index: 200;"
  >
    <div>
      <svg
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="c0"
        height="48"
        preserveAspectRatio="xMidYMid meet"
        role="progressbar"
        viewBox="24 24 48 48"
        width="48"
      >
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_13, rgba(222, 237, 238, 1))"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
        <circle
          class=""
          cx="48"
          cy="48"
          fill="none"
          r="22"
          stroke="var(--eds_infographic_primary__moss_green_100, rgba(0, 112, 121, 1))"
          stroke-dasharray="138.23007675795088"
          stroke-linecap="round"
          stroke-width="4"
          style="stroke: 138.230; stroke-dashoffset: 138.230px;"
        />
      </svg>
      <output
        class="c1"
      >
        Loading 0%
      </output>
      <br />
      Loading assets...
    </div>
  </div>
</div>
`;
